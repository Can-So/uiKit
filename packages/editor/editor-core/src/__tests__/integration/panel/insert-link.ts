import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';

import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
} from '../_helpers';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import { selectors } from './_utils';

BrowserTestCase(
  'insert-link.ts: Insert link in panel by typing Markdown',
  { skip: ['edge', 'ie'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowPanel: true,
    });

    await page.click(fullpage.placeholder);

    await quickInsert(page, 'Panel');
    await page.waitForSelector(selectors.PANEL_EDITOR_CONTAINER);

    await page.type(editable, '[Atlassian](https://www.atlassian.com/)');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);
