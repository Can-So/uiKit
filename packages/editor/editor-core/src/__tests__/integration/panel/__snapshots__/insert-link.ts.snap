// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert-link.ts: Insert link in panel by typing Markdown 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "marks": Array [
                Object {
                  "attrs": Object {
                    "__confluenceMetadata": null,
                    "href": "https://www.atlassian.com/",
                  },
                  "type": "link",
                },
              ],
              "text": "Atlassian",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;
